### ISSUE TITLE
Improve Stacking

### ISSUE THREAD

**Issue by giuseppec** [mlr-org/mlr#702]
As discussed in https://github.com/mlr-org/mlr/issues/697 we need to change the stacking functions. Maybe something like:

1) makeStackedTask: should create a new task that contains the features (eg. only the CV predictions, CV predictions + original features... ) for stacking. Also we should allow repeated CV predictions (currently this is not supported).

2) building the superlearner should than work by training the superlearner on the task  from 1). One could also do resampling to estimate the superlearner performance. 

Problem : if we use "stacknocv" method, one could not trust the superlearner performance because of overfitting. However, nested CV could help out. 

General advantage : preprocessing after step 1) is possible.

**Comment by berndbischl** [comment]
3) The hyperpars of the super.learner should be in the param set of the stacked learner. this is most simple and should be done first. 

For 1) and 2) please post signature / API suggestions here before coding this.

**Comment by giuseppec** [comment]
Here a very coarse pseudo-code:

```r
#' @param base.learners: list of learners
#' @param task: your task with training data
#' @param resampling: a resample desc or resample instance (or a list of resample instances, where the list has the same length as base.learners, so that different obs are used for the cv-predictions of each base learner)
#' @method: either stackCV, stackNoCV, average....
#' @param use.feat: should original features from task also be used?
stackTask = makeStackingTask(base.learners, task, resampling, method, use.feat = FALSE) {
 if(method != "stackNoCV") {
  # get cv-predictions from all base learners
  cvPreds = lapply(base.learners, function(x) resample(x, task, resampling = resampling) )
  # note1: order the cv-prediction according to their observation 'id'
  # note2: if resample is repeated-cv, just take the average of all replications for the same obs (bootstrap not supported)
  if(use.feat) {
    # make and return task using cvPreds and features from 'task'
  } else {
    # make and return task using cvPreds (without features from 'task')
  }
 }
}
```
Got the idea? So we could apply a superlearner on this new "level 1" task and tuning works as usual.

**Comment by berndbischl** [comment]
So it takes certain base.learners, which have parameters maybe changed, and return a task which contains their predictions?

I guess I like that

**Comment by berndbischl** [comment]
this seems outdated, we have PR #1041 now
